File: content-solutions/implementation/sim-turn-walk.js
function solution(commands) {
  const dx = [0, 1, 0, -1];
  const dy = [1, 0, -1, 0];
  let x = 0;
  let y = 0;
  let d = 0;
  for (const ch of commands) {
    if (ch === "L") d = (d + 3) % 4;
    else if (ch === "R") d = (d + 1) % 4;
    else {
      x += dx[d];
      y += dy[d];
    }
  }
  return [x, y];
}

File: content-solutions/implementation/sim-turn-walk.py
def solution(commands):
    dx = [0, 1, 0, -1]
    dy = [1, 0, -1, 0]
    x = y = d = 0
    for ch in commands:
        if ch == "L":
            d = (d + 3) % 4
        elif ch == "R":
            d = (d + 1) % 4
        else:
            x += dx[d]
            y += dy[d]
    return [x, y]
